### ISSUE TITLE
Tooltip `content` does not provide index of which bar is active or being hovered on

### ISSUE THREAD

**Issue by Andriy-Kulak** [recharts/recharts#3835]
- [x] I have searched the [issues](https://github.com/recharts/recharts/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[![Edit on CodeSandbox](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/tooltip-with-customized-content-forked-n4k66r?file=/src/App.tsx)

### Steps to reproduce
look at the console.log of props. We need to expose the index of which bar chart or x axis is being hovered on or active. It makes it impossible to provide tooltip with extra data because i don't know which bar chart is being hovered on

### What is expected?
provide an index of bar chart into `content` props of tooltop

### What is actually happening?
no index is provided

| Environment | Info |
|---|---|
| Recharts | v2.8.0 |
| React | 18.2.0 |
| System | any system |
| Browser | any browser |

**Comment by zachterrell57** [comment]
+1, also having this problem

**Comment by nikolasrieble** [comment]
This is a great feature request, and it could possibly improve performance. @andrewangelle If we had the index, we would not have to compare content for finding the original data point for the Tooltip, right? 

The solution is quite simple, we could use a https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Symbol to set another field per row in the original dataset, and then use the symbol everywhere internally. 

In order for users to utilise this as well, we would have to export the symbol as well.

**Comment by andrewangelle** [comment]
@nikolasrieble You wanna assign me? I'll take a look at implementing this next.

**Comment by Andriy-Kulak** [comment]
thanks @andrewangelle & @nikolasrieble for prompt reply. any idea on when do you think this will be in a release of library? just curious so I have an idea of when to check back in.

**Comment by ckifer** [comment]
Time to get this merged and release will be quite variable because this is all spare/volunteer time. We're planning on a release soon but there is a good chance this won't be included until the one after (maybe 1 month+ from now). 

Following the issue should alert you when it's updated

**Comment by Andriy-Kulak** [comment]
> Time to get this merged and release will be quite variable because this is all spare/volunteer time. We're planning on a release soon but there is a good chance this won't be included until the one after (maybe 1 month+ from now).
> 
> Following the issue should alert you when it's updated

@ckifer @andrewangelle - thanks. any ideas for a temp workaround of how to know which bar chart is being hovered on within a tooltip? I cannot use `label` because that's not a unique property in my scenario. Our bug basically happens when a user adds 2 of the same events in chart builder & they therefore now have the same label name on the x axis.

**Comment by ckifer** [comment]
@Andriy-Kulak this might work quite well 

https://codesandbox.io/s/simple-bar-chart-forked-tk9c6z?file=/src/App.tsx

`onMouseMove` gets some info about the tooltip including activeTooltipIndex. You can store that and pass it down to your custom tooltip

**Comment by GuyMichael** [comment]
Any update on this feature request? Would be amazing to have it

**Comment by ckifer** [comment]
Been a while since I've looked at this so will provide my current thoughts:

The interface for "active" things is not the same across all charts. It can't be because some charts don't have normal indexes (nested data in treemap or Sankey or sunburst, etc.). That said, I would rather provide the "active data" (payload, what we currently do) than provide an active index to Tooltip which is component used across all charts. It would be a useless field for the things that don't easily support indices

---

But with that said I'd like to understand the use-case better. What's preventing you from using active index from mouse events or using the payload itself to find extra data?

**Comment by GuyMichael** [comment]
> The interface for "active" things is not the same across all charts
> 
> But with that said I'd like to understand the use-case better. What's preventing you from using active index from mouse events or using the payload itself to find extra data?

That's a valid input, thx. It does make it "weird" to have it for only a few chart types.
It's actually a no-brainer -I just added an "__index" to my data so that I'd get it in the (Tooltip's) payload.
I don't want to use mouse events because (in React, for example, but not limited to) I'd need to have a (use)State on the chart Component, making it re-render on every mouse-move (so that the Tooltip would receive the updated activeIndex).

My use-case is a MagnifierTooltip, which receives the whole data and the hovered index, in order to render a Tooltip with a secondary (zoom-in, containing just a few points around the hovered point) chart

**Comment by ckifer** [comment]
It might work to include `index` in the `payload` object rather than the overall `content` object.

But in that case `index` becomes a reserved word and can never exist in a user's data

**Comment by GuyMichael** [comment]
yeah, that's a bit weird, but I think finding a valid approach would make it easier to further advance with other widgets in the library (suggestion below: `__meta`).

For example, the `ReferenceArea` receives values/labels rather than indices (please correct me if I'm wrong!) which makes it necessary for me to 'inject' another field to my dataset (say, 'name', or 'label', for passing it on to the `ReferenceArea`).

So anyway, I think working with indices makes life easier (e.g. no label collision).

So.... What about having a `__meta` object in the `payload`, which will contain `index`, among other things which might be helpful in the future. And yes, that would make `__meta` a reserved word. Not that terrible though, right?

**Comment by PavelVanecek** [comment]
New property is added in https://github.com/recharts/recharts/pull/6586 and will go out in the next release.
